internal/internal_worker_base.go

```diff
@@ -54,6 +57,8 @@ var (
 
 var errShutdown = errors.New("worker shutting down")
 
+var emitOnce sync.Once
```

Comment:
I would suggest extracting this into WorkerResult as a separate dependency, instead of trying to fit into the workers with sync.Once.

Reply:
What do you mean by WorkerResult? Are you referring to the baseWorker struct?

Reply:
Oh well, I thought we provided fx.Module for this as we do in the monorepo. There is a NewCadenceWorkers that returns a single result that can be a good place to have this component.
If we want to care about monorepo - then it should be part of monorepo.
If we want to have this in OSS, then, I think, it should be a separated component that is optional and could be used by developers as an extra option.

Reply:
I'm not even sure that we want to have this in OSS, since a developer could use something like https://github.com/prometheus/client_golang/blob/main/prometheus/go_collector.go#L216
So it will expose data for collection.

Reply:
If we want to do it in OSS, we cannot use the package that only works for prometheus. Cadence use tally on top of underlying metrics package.